### PR TITLE
feat(unlock-app): Stop asking user to sign in repeatedly

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

This fixes an issue where we repeatedly ask user to sign in. It's a temporary fix which I will remove in rewrite. 

Please look carefully if this breaks any of your code (I've minimized the change so it shouldn't) @kalidiagne 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — APPROVED** [review]
Works for me, but please @kalidiagne let us know if it works for you too!

**Comment by kalidiagne** [comment]
@searchableguy were to do you see repeated sign-in popup? in the current version sign-in will be shown only when wallet account is changed or the token expired, is it not? 

Testing with this branch, i see an issue for example in `verifiers page`, when i look at the verifiers page and change account from Metamask I'm able to see the list even if I'm not the lock manager because the token is still the same and a new pop-up request to sign-in does not show when I reload the page.

**Comment by kalidiagne** [comment]
@searchableguy added a commit that solve the issue described before, we need to remove access and refresh token when  wallet account changes

File: unlock-app/src/hooks/useProvider.ts [hunk 1]
```diff
@@ -117,8 +117,10 @@ export const useProvider = (config: any) => {
       }
       const ethersProvider = new ethers.providers.Web3Provider(provider)
 
+      // remove access token when account wallet changes
       const removeAccessToken = () => {
-        localStorage.removeItem(`${APP_NAME}.token`)
+        localStorage.removeItem(`locksmith-access-token`)
```
Comment (kalidiagne) [inline unlock-app/src/hooks/useProvider.ts:122]: @searchableguy maybe store this name as done before?

**Comment by searchableguy** [comment]
That case should be handled by logging out the user when they disconnect an account. You would need to call the logout endpoint and remove the tokens. I've not seen anything implemented to that extent already.

Want me to do it? @kalidiagne

**Comment by searchableguy** [comment]
Logging out wipes out the token from backend so it's an extra precaution in case of any possible leaks.

**Comment by searchableguy** [comment]
@kalidiagne can you re-test and see if all issues are resolved?

I will add the logout call later when I refactor it. For now, should be okay.

**Comment by searchableguy** [comment]
@kalidiagne re-test and see if there are any problems?

File: unlock-app/src/services/storageService.ts [hunk 2]
```diff
@@ -71,13 +70,27 @@ export class StorageService extends EventEmitter {
     return this.locksmith.login(message, signature)
   }
 
+  async signOut() {
+    const endpoint = `${this.host}/v2/auth/revoke`
+    const response = await fetch(endpoint, {
```
Comment (julien51) [inline unlock-app/src/services/storageService.ts:75]: Let's make this fail silently (try/catch) since it should not affect the rest of the app if it does.

**Comment by kalidiagne** [comment]
@searchableguy tested and all fine!

**Review by kalidiagne — APPROVED** [review]

**Pull request merged**